service/discovery/aws/aws.go

```diff
@@ -39,20 +42,52 @@ var log = logrus.WithField("component", "aws-discovery")
 // loadDefaultConfig holds config.LoadDefaultConfig() so the test function can mock it
 var loadDefaultConfig = config.LoadDefaultConfig
 
+// newFromConfigSTS holds sts.NewFromConfig() so the test function can mock it
+var newFromConfigSTS = loadSTSClient
+
 // Client holds configurations across all services within AWS
 // TODO(lebogg): deepsource.io wants the struct to exported since NewAwsStorageDiscovery is exported. Encapsulation?
 type Client struct {
 	Cfg aws.Config
```

Comment:
Any particular reason why the client Cfg is exposed?

Reply:
Not really. In the package `discovery` we cannot access it then. But it is not needed. I changed it and the respective lines of code.